# Тести для малих вибірок та пропорцій



**Завдання 1**. E-commerce компанія після редизайну сайту підозрює, що **середній час до покупки (time-to-purchase)** користувачів **збільшився**.
Історично середній час від першого заходу на сайт до покупки становив $\mu_0 = 123.7$ хвилин.

Команда аналітиків випадково обрала дані **7 покупців після редизайну**:
`128, 135, 121, 142, 126, 151, 123`

З допомогою наявних даних зʼясуйте, чи збільшився середній час покупки після редизайну.

Для цього виконайте наступні 6 кроків. Правильне виконання кожного з кроків оцінюється в 1 бал.

1. Запишіть нульову та альтернативну гіпотези і визначте тип тесту.
2. Обчисліть вибіркові статистики: $\bar x$, $s$, $n$.
3. Оберіть тип тесту та виконайте його (знайдіть критичне значення тесту, статистику тесту та р-значення) будь-яким пасуючим способом, щоб перевірити гіпотезу на рівні значущості $\alpha = 0.10$.
4. Прийміть рішення, чи відхиляєте ви гіпотезу $H_0$ використовуючи p-value.
5. Напишіть висновок: чи справді редизайн сайту подовжив час до покупки?
6. Чи зміниться ваше рішення при зміні рівня значущості на $\alpha = 0.05$.


H₀: μ = 123.7
Hₐ: μ > 123.7
Тип тесту правосторонній, адже при альтернатівній гіпотезі вважаємо, що середній час покупки збільшився.

In [ ]:
import numpy as np
from scipy import stats

In [ ]:
average_purchase_time= np.array([128, 135, 121, 142, 126, 151, 123])
mu0 = 123.7
alpha = 0.1


In [ ]:
sample_mean = np.mean(average_purchase_time)
sample_std = np.std(average_purchase_time, ddof=1)
n = len(average_purchase_time)
print(sample_mean, sample_std, n)

132.28571428571428 10.980502200328935 7


In [ ]:
# обчислюю t-статистику вручну
t_stat = (sample_mean - mu0) / (sample_std / np.sqrt(n))
print(t_stat)

2.0687273144186475


In [ ]:
df = n - 1
t_critical = stats.t.ppf(1 - alpha, df)
print(t_critscal)

1.4397557472577693


In [ ]:
p_value = 1 - stats.t.cdf(t_stat, df)
print(p_value)

0.042017654253682646


In [ ]:
print(f"Вибіркове середнє: {sample_mean:.2f}")
print(f"Вибіркове стандартне відхилення: {sample_std:.2f}")
print(f"t-статистика: {t_stat:.3f}")
print(f"Критичне значення (df={df}): {t_critical:.3f}")
print(f"p-value: {p_value:.4f}")

Вибіркове середнє: 132.29
Вибіркове стандартне відхилення: 10.98
t-статистика: 2.069
Критичне значення (df=6): 1.440
p-value: 0.0420


In [ ]:
print("Висновок:")

if t_stat > t_critical:
    print(f"Відхиляємо H0: середній час збільшився на рівні значущості {alpha}")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Висновок:
Відхиляємо H0: середній час збільшився на рівні значущості 0.1


Редизайн сайту подовжив середній час покупки. Середній час покупки збільшився із 123.7  хвилин до 132.29 хвилин.

Після зміни рівня значущості на 0.05 рішення не зміниться, оскільки p-value 0.0420 менше за 0.05, нульова гіпотеза
 відхиляється.

**Завдання 2.**

До спеціальної рекламної кампанії **23%** дорослих упізнавали логотип компанії. Після завершення кампанії відділ маркетингу провів опитування: з **1200** випадково відібраних дорослих **311** упізнали логотип.

Перевірте на рівні значущості **3%** ($\alpha=0.03$), чи дають ці дані достатні підстави стверджувати, що **тепер більше ніж 23%** дорослих упізнають логотип компанії. Для розвʼязку використовуйте бібліотеку `statsmodels`.

Зробіть висновок, чи зросла впізнаваність логотипу.

Додатково, обчисліть довірчий інтревал на заданому рівні значущості і проінтерпретуйте текстом - як він додатково пояснює прийняте нами рішення?

H₀: p = 0.23
Hₐ: p > 0.23
Це правосторонній тест, осткільки є альтернативна гіпотеза, що бульше ніж 23% дорослих упізнають логотип компанії.

In [ ]:
from statsmodels.stats.proportion import proportions_ztest, proportion_confint

In [ ]:
count = 311
nobs = 1200
p0 = 0.23
alpha = 0.03



In [ ]:
z_stat, p_value = proportions_ztest(count, nobs, value=p0, alternative='larger')

In [ ]:
confint = proportion_confint(count,
                             nobs,
                             alpha,
                             method='normal')

In [ ]:
print("Тест для пропорції впізнаваності")
print(f"Z-статистика = {z_stat:.3f}")
print(f"p-value = {p_value:.4f}")
print(f"97% довірчий інтервал: ({confint[0]:.3f}, {confint[1]:.3f})")

Тест для пропорції впізнаваності
Z-статистика = 2.306
p-value = 0.0106
97% довірчий інтервал: (0.232, 0.287)


In [ ]:
print("Висновок:")
if p_value < alpha:
    print(f"Відхиляємо H0 на рівні значущості {alpha}")
    print("Є статистично значущі підстави вважати, що пропорція збільшилась")
else:
    print("Не відхиляємо H0: доказів недостатньо")

Висновок:
Відхиляємо H0 на рівні значущості 0.03
Є статистично значущі підстави вважати, що пропорція збільшилась
